libs/books/data-access/src/lib/+state/reading-list.effects.spec.ts

```diff
@@ -42,4 +44,56 @@ describe('ToReadEffects', () => {
       httpMock.expectOne('/api/reading-list').flush([]);
     });
   });
+
+  describe('addBook$', () => {
+    it('should invoke EMPTY action when addToReadingList action succeeds', () => {
+      actions = new ReplaySubject();
+      actions.next(ReadingListActions.addToReadingList({
+        book: createBook('A'), showSnackBar: false
+      }));
+
+      effects.addBook$.subscribe(action => {
+        expect(action).toEqual(EMPTY);
+      });
+
+      httpMock.expectOne('/api/reading-list').flush([]);
+    });
+
+    it("shoul call openSnackBar function when showSnackBar is true", () => {
+      jest.spyOn(effects, 'openSnackBar');
+      actions = new ReplaySubject();
+      actions.next(ReadingListActions.addToReadingList({
+        book: createBook('A'), showSnackBar: true
+      }));
+
+      effects.addBook$.subscribe();
+      expect(effects.openSnackBar).toHaveBeenCalledWith('Added !')
+    });
+  });
+
+  describe('removeBook$', () => {
+    it('should invoke EMPTY action when removeFromReadingList action succeeds', () => {
+      actions = new ReplaySubject();
+      actions.next(ReadingListActions.removeFromReadingList({
+        item: createReadingListItem('A'), showSnackBar: false
+      }));
+
+      effects.removeBook$.subscribe(action => {
+        expect(action).toEqual(EMPTY);
```

Comment:
There's a chance this never gets called so the expect isn't run.

Use the `done` callback and make sure to call it whenever async expectations are done in a callback function.

```
it('should ...', (done) => {

doSomething.then(() => {
  expect(...).toEqual(...);
  done(); // this will fail if not called
});
```